pkgs/applications/science/logic/isabelle/default.nix

```diff
@@ -57,6 +63,12 @@ in stdenv.mkDerivation rec {
     sourceRoot=${dirname}
   '' else null;
 
+  prePatch = if hgSrc == null then null else ''
+    rm -r src
+    cp -r ${hgSrc}/src ./
+    chmod -R +w ./src
+  '';
```

Comment:
Can we avoid this null checkery, too?

Also this fits more into postUnpack before patchPhase.